src/sage/combinat/integer_vector.py

```diff
@@ -839,7 +838,10 @@ def __init__(self, n):
             sage: TestSuite(IV).run()
         """
         self.n = n
-        IntegerVectors.__init__(self, category=InfiniteEnumeratedSets())
+        if self.n==0:
```

Comment:
There need to be spaces around `==`.  Alternatively (and likely more according to the unwritten sage conventions, but I don't care):
```
if self.n:
    IntegerVectors.__init__(self, category=InfiniteEnumeratedSets())
else:
    IntegerVectors.__init__(self, category=EnumeratedSets())